app/models/observation.rb

```diff
@@ -193,6 +193,50 @@ class Observation < AbstractModel
   # Automatically (but silently) log destruction.
   self.autolog_events = [:destroyed]
 
+  include CreatedUpdatedScopes
+
+  scope :include_all_name_proposals, lambda { |name|
+    joins(:namings).where(namings: { name: name })
+  }
+  scope :of_look_alikes, lambda { |name|
+    joins(:namings).where(namings: { name: name }).
+      where(Observation[:name] != name)
+  }
+  scope :include_subtaxa, lambda { |text_name|
+    names = {}
+    starting_names = Name.text_name_like(text_name).with_correct_spelling
+    starting_names.each do |name|
+      names = starting_names.or(Name.related_taxa(name)).distinct
+    end
+    where(name: names)
+  }
+  scope :of_synonyms, ->(name) { where(name: name.synonyms) }
+  scope :of_name, ->(name) { where(name: name) }
+  scope :of_name_like,
+        ->(name) { where(name: Name.text_name_like(name)) }
+  scope :with_name, -> { where.not(name: Name.unknown) }
+  scope :without_name, -> { where(name: Name.unknown) }
+  scope :by_user, ->(user) { where(user: user) }
+  scope :at_location, ->(location) { where(location: location) }
+  scope :in_region,
+        ->(where) { where(Observation[:where].matches("%#{where}%")) }
```

Comment:
Nice!  This one really does seem worthwhile to me.

Reply:
Ahaa. Yes that sounds right. Does that hold true in all languages?